torch/fx/_symbolic_trace.py

```diff
@@ -430,9 +436,17 @@ def call_module(
             value was returned from the ``Module`` invocation.
         """
         module_qualified_name = self.path_of_module(m)
-        if not self.is_leaf_module(m, module_qualified_name):
-            return forward(*args, **kwargs)
-        return self.create_proxy("call_module", module_qualified_name, args, kwargs)
+        with ScopeContextManager(self.scope, m, module_qualified_name) as _scope:
+            # module_stack is an ordered dict so writing then deleting the
+            # entry is equivalent to push/pop on a list
+            self.module_stack[_scope.module_path] = str(_scope.module_type)
```

Comment:
actually, it feels like we can probably maintain this thing in a scope as well

Reply:
The self.module_stack is the data structure that populates node.meta['nn_module_stack'].  It is supposed to be an OrderedDict to be consistent with torch dynamo logic that @SherlockNoMad implemented.